gcovr/formats/gcov/parser.py

```diff
@@ -512,7 +512,9 @@ def _gather_coverage_from_line(
     elif isinstance(line, _FunctionLine):
         # Defer handling of the function tag until the next source line.
         # This is important to get correct line number information.
-        return state._replace(deferred_functions=[*state.deferred_functions, line])
+        return state._replace(
+            deferred_functions=[*state.deferred_functions, line]
+        )._replace(blockno=0)
```

Comment:
Can replace multiple fields in one call.

```suggestion
        return state._replace(
            deferred_functions=[*state.deferred_functions, line],
            blockno=0,
        )
```

However, it might be preferable to apply the block reset at the point where the deferred function is applied?

Reply:
I`ll undo the changes in the parser and add a property to the data model to get always a valid blockno.